File: 2317-count-collisions-on-a-road/2317-count-collisions-on-a-road.cpp
class Solution {
public:
    int countCollisions(string directions) {
       stack<char> st;
       int ans = 0;
       for(int i = 0 ; i < directions.size() ; i++){
            st.push(directions[i]);
            while(st.size()>1){
                char current = st.top();
                st.pop();
                char prev = st.top();
                st.pop();
                if(current == 'L' && prev == 'R'){
                    ans+=2;
                    st.push('S');
                    continue;
                }else if(current == 'S' && prev =='R' ){
                    ans+=1;
                    st.push('S');
                    continue;

                }else if(current == 'L' && prev == 'S' ){
                    ans+=1;
                    st.push('S');
                    continue;

                }else{
                    st.push(prev);
                    st.push(current);
                    break;
                }
            }
            
       }
       return ans;
    }
};

File: 2317-count-collisions-on-a-road/2317-count-collisions-on-a-road.java
class Solution {
    public int countCollisions(String directions) {
        Stack<Character> st = new Stack<>();
        int count = 0;
        for(int i=  0 ; i < directions.length() ; i++){
            Character ch = directions.charAt(i);
            st.push(ch);
            while(st.size()>1){
                Character current = st.pop();
                Character prev = st.pop();
                if(current == 'L' && prev =='R'){
                    count+=2;
                    st.push('S');
                }else if(current == 'S' && prev == 'R'){
                    count++;
                    st.push('S');
                }else if(current == 'L' && prev == 'S'){
                    count++;
                    st.push('S');
                }else{
                    st.push(prev);
                    st.push(current);
                    break;
                }
                
            }
        }
        return count;
    }
}